src/TestUtils/src/Microsoft.Maui.IntegrationTests/Utilities/BuildWarningsUtilities.cs

```diff
@@ -0,0 +1,632 @@
+							}
+						}
+				};
+				warnings.Add(newEntry);
+			}
+			else
+			{
+				var warningsPerCode = warningsPerFile.WarningsPerCode.FirstOrDefault(w => w.Code == code);
+				if (warningsPerCode is null)
+				{
+					var newEntry = new WarningsPerCode
+					{
+						Code = code,
+						Messages = new List<string> { message }
+					};
+					warningsPerFile.WarningsPerCode.Add(newEntry);
+				}
+				else
+				{
+					warningsPerCode.Messages.Add(message);
+				}
+			}
+		}
+
+		public static void AssertWarnings(this List<WarningsPerFile> actualWarnings, List<WarningsPerFile> expectedWarnings)
+		{
+			foreach (var expectedWarningsPerFile in expectedWarnings)
+			{
+				var actualWarningsPerFile = actualWarnings.FirstOrDefault(actualWarning => actualWarning.File.CompareWarningsFilePaths(expectedWarningsPerFile.File));
+				Assert.NotNull(actualWarningsPerFile,
+					$"Expected warnings file path '{expectedWarningsPerFile.File}' was not found.");
+
+				foreach (var expectedWarningsPerCode in expectedWarningsPerFile.WarningsPerCode)
+				{
+					var actualWarningsPerCode = actualWarningsPerFile!.WarningsPerCode.FirstOrDefault(x => x.Code == expectedWarningsPerCode.Code);
+					Assert.NotNull(actualWarningsPerCode,
+						$"Expected warning code '{expectedWarningsPerCode.Code}' was not found for the expected warnings file path '{expectedWarningsPerFile.File}'");
+
+					foreach (var expectedWarningsMessage in expectedWarningsPerCode.Messages)
+					{
+						Assert.True(actualWarningsPerCode!.Messages.Remove(expectedWarningsMessage),
+							$"Expected warning message '{expectedWarningsMessage}' was not found for the expected warnings file path '{expectedWarningsPerFile.File}' and warning code '{expectedWarningsPerCode.Code}'");
+					}
+
+					Assert.AreEqual(0, actualWarningsPerCode!.Messages.Count,
+						$"Unexpected warning messages detected for the expected warnings file path '{expectedWarningsPerFile.File}' and warning code '{expectedWarningsPerCode.Code}'! Unexpected warning messages are: {string.Join("\n\t\t", actualWarningsPerCode.Messages)}");
+
+					actualWarningsPerFile.WarningsPerCode.Remove(actualWarningsPerCode);
+				}
+
+				Assert.AreEqual(0, actualWarningsPerFile!.WarningsPerCode.Count,
+					$"Unexpected warning codes detected for the expected warnings file path '{expectedWarningsPerFile.File}'! Unexpected warning codes are: {string.Join("\n\t\t", actualWarningsPerFile.WarningsPerCode.Select(c => c.Code).ToList())}");
+
+				actualWarnings.Remove(actualWarningsPerFile!);
+			}
+
+			Assert.AreEqual(0, actualWarnings.Count,
+				$"Unexpected warning files detected! Unexpected warning file paths are: {string.Join("\n\t\t", actualWarnings.Select(f => f.File).ToList())}");
+		}
+
+		#region Expected warning messages
+
+		// IMPORTANT: Always store expected File information as a relative path to the repo ROOT
+		private static readonly List<WarningsPerFile> expectedNativeAOTWarnings = new()
```

Comment:
I was thinking a `.txt` file with a line per warning, but this approach seems OK.

If our goal is to get this to 0 warnings, we'll eventually delete `BuildWarningsUtilities.cs` anyway?

Reply:
The end goal is to reach `expectedNativeAOTWarnings = new()`.
While we might want to keep `BuildWarningsUtilities.cs` if we start checking the same for Mono for example.